### PR TITLE
Update default efSearch and fix pagination bug 

### PR DESCRIPTION
**Pull request by farshidz**

### CODE REVIEW

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py [hunk 1]
```diff
@@ -314,17 +314,25 @@ def _get_tensor_search_term(self, marqo_query: MarqoTensorQuery) -> str:
...
-                f'({{targetHits:{marqo_query.limit}, approximate:{str(marqo_query.approximate)}{additional_hits}}}'
-                f'nearestNeighbor({embedding_field_name}, {common.QUERY_INPUT_EMBEDDING}))'
+                f'('
+                f'{{'
+                f'targetHits:{target_hits}, '
+                f'approximate:{str(marqo_query.approximate)}, '
+                f'hnsw.exploreAdditionalHits:{additional_hits}'
+                f'}}'
+                f'nearestNeighbor({embedding_field_name}, {common.QUERY_INPUT_EMBEDDING})'
+                f')'
```
Comment (pandu-k) [inline src/marqo/core/structured_vespa_index/structured_vespa_index.py:337]: What about the case where limit = 10, offset = 1000, and efsearch=512
Won't target + additional=512? Don't we need a bigger efsearch to get further into the hit list?

File: src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py
Comment (pandu-k) [inline src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py:1]: ditto above comment

File: src/marqo/tensor_search/configs.py [hunk 2]
```diff
@@ -29,5 +29,6 @@ def default_env_vars() -> dict:
         EnvVars.MARQO_MAX_DELETE_DOCS_COUNT: 10000,
         EnvVars.MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES: None,
         EnvVars.MARQO_MAX_NUMBER_OF_REPLICAS: 1,
+        EnvVars.MARQO_DEFAULT_EF_SEARCH: 2000,
```
Comment (pandu-k) [inline src/marqo/tensor_search/configs.py:32]: are we happy with the throughput at this level?

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py (hunk 1 above)
Comment (pandu-k) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py:337]: Ok I see there is search logic ensuring that efSearch is always >= offset + limit

Comment (pandu-k) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py:337]: we need to ensure this is clearly specified in documentation

**Review by pandu-k — APPROVED** [review]

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py (hunk 1 above)
Comment (farshidz) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py:337]: Yes if efSearch is too low they'll get empty hits

File: src/marqo/tensor_search/configs.py (hunk 2 above)
Comment (farshidz) [reply src/marqo/tensor_search/configs.py:32]: We are. Performance drop was minor

**Pull request merged**